src/main/java/org/checkerframework/checker/builder/autovalue/AutoValueBuilderAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,162 @@
+package org.checkerframework.checker.builder.autovalue;
+
+import com.google.auto.value.AutoValue;
+import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.Tree;
+import java.lang.annotation.Annotation;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Modifier;
+import org.checkerframework.checker.builder.CalledMethodsAnnotatedTypeFactory;
+import org.checkerframework.checker.builder.CalledMethodsUtil;
+import org.checkerframework.checker.builder.TypesafeBuilderQualifierHierarchy;
+import org.checkerframework.checker.builder.qual.CalledMethods;
+import org.checkerframework.checker.builder.qual.CalledMethodsBottom;
+import org.checkerframework.checker.builder.qual.CalledMethodsPredicate;
+import org.checkerframework.checker.builder.qual.CalledMethodsTop;
+import org.checkerframework.com.google.common.collect.ImmutableSet;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.framework.util.MultiGraphQualifierHierarchy;
+import org.checkerframework.framework.util.MultiGraphQualifierHierarchy.MultiGraphFactory;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+
+/** Responsible for placing appropriate annotations on Lombok builders. */
+public class AutoValueBuilderAnnotatedTypeFactory extends BaseAnnotatedTypeFactory
```

Comment:
This is the part I am most conflicted about.  I understand we want to have built-in framework support be configurable, and doing it via separate type factories kind of makes sense.  But in terms of code structure it is a mess, particularly because it seems the `TypesafeBuilderAnnotatedTypeFactory` cannot get these inserted annotations "for free" and has to again re-insert them.  Maybe the code is just wrong?  I think we should consider alternatives for configuring framework support, such as configurable options within the type safe builder checker.  That won't let users plug in their own framework support, but properly supporting that would be a bunch more work too (I think).